Call FontCache::invalidate under memory pressure

heycam opened the pull request:
#### db0ade3c62fb2c65e7b2b1a8060b886981ded622
<pre>
Call FontCache::invalidate under memory pressure
<a href="https://bugs.webkit.org/show_bug.cgi?id=249561">https://bugs.webkit.org/show_bug.cgi?id=249561</a>
rdar://103498661

Reviewed by Myles C. Maxfield.

The changes in bug 234061 caused us not to call FontCache::invalidate
under memory pressure, which we should do, since it clears some data
that the current platformReleaseNoncriticalMemory does not do.

The two things that are done -- clearing m_systemFontDatabaseCoreText
and m_fontFamilySpecificationCoreTextCache -- are also done by
invalidate(), so we can replace them with an invalidate() call.

* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::FontCache::platformReleaseNoncriticalMemory):

Canonical link: <a href="https://commits.webkit.org/258443@main">https://commits.webkit.org/258443@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a5cefe650fa7fb7335f7e0d55674fa3cbaa8a1f8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101891 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11038 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34963 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111224 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/171434 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/105871 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12001 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1953 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94297 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108979 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107672 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9182 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92450 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/36977 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/91067 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/23880 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/78743 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4623 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25364 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4710 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1801 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10788 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44851 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6462 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3048 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

litherum reviewed (approved):
r=me assuming this doesn't cause infinite recursion

heycam committed:
Call FontCache::invalidate under memory pressure
https://bugs.webkit.org/show_bug.cgi?id=249561
rdar://103498661

Reviewed by Myles C. Maxfield.

The changes in bug 234061 caused us not to call FontCache::invalidate
under memory pressure, which we should do, since it clears some data
that the current platformReleaseNoncriticalMemory does not do.

The two things that are done -- clearing m_systemFontDatabaseCoreText
and m_fontFamilySpecificationCoreTextCache -- are also done by
invalidate(), so we can replace them with an invalidate() call.

* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::FontCache::platformReleaseNoncriticalMemory):

Canonical link: https://commits.webkit.org/258443@main
